File: backend/src/main/java/com/back/domain/bookmarks/dto/BookmarkBookDetailDto.java
package com.back.domain.bookmarks.dto;

import com.back.domain.book.book.entity.Book;

import java.time.LocalDateTime;
import java.util.List;

public record BookmarkBookDetailDto(
        int id,
        String isbn13,
        String title,
        String imageUrl,
        String publisher,
        int totalPage,
        double avgRate,
        String category,
        List<String> authors,
        LocalDateTime publishDate
) {
    public BookmarkBookDetailDto(Book book){
        this(
                book.getId(),
                book.getIsbn13(),
                book.getTitle(),
                book.getImageUrl(),
                book.getPublisher(),
                book.getTotalPage(),
                book.getAvgRate(),
                book.getCategory().getName(),
                book.getAuthors() !=null ? book.getAuthors().stream().map(a -> a.getAuthor().getName()).toList() : List.of(),
                book.getPublishedDate()
        );
    }
}

File: backend/src/main/java/com/back/domain/bookmarks/dto/BookmarkBookDetailDto.kt
package com.back.domain.bookmarks.dto

import com.back.domain.book.book.entity.Book
import com.back.domain.book.wrote.entity.Wrote
import java.time.LocalDateTime


data class BookmarkBookDetailDto(
    val id: Int,
    val isbn13: String?,
    val title: String?,
    val imageUrl: String?,
    val publisher: String?,
    val totalPage: Int,
    val avgRate: Double,
    val category: String?,
    val authors: List<String>,
    val publishDate: LocalDateTime?
) {
    constructor(book: Book) : this(
        book.id,
        book.isbn13,
        book.title,
        book.imageUrl,
        book.publisher,
        book.totalPage,
        book.avgRate.toDouble(),
        book.category.name,
        book.authors.map { a: Wrote -> a.author.name },
        book.publishedDate
    )
}
